mantidproject/mantid
# Re #19951:  ISIS Reflectometry: Prevent external table modification while processing

- Disables actions which modify the contents of the DataProcessorWidget runs table whilst the widget is processing and not paused.
- Refactors elements of the `QDataProcessorWidget`, `QtReflRunsTabView`, `ReflRunsTabPresenter` and `GenericDataProcessor` as well as the tests `ReflRunsTabPresenterTest`.

**To test:**
- Unit Tests:
  - Compile and run the test targets `MantidWidgetsTest` and `CustomInterfacesTest`
- Manual Tests:
  - Login into the ICAT catalog.
  - The interface can be accessed from Interfaces->Reflectometry->ISIS Reflectometry.
  - Select `POLREF`, enter investigation id `1520249` and click on the `autoreduce` button to begin autoreduction.
  - Ensure that it is not possible to perform actions which edit the contents of table while processing through either the Data Processor Widget toolbar, the context (right click) menu, or the 'Edit' and 'Reflectometry' menus.
  - Ensure that the actions disabled while processing are made available again when the processing is paused and when the processing is completed.

Fixes #19951. 

**Release Notes** 
*Does not need to be in the release notes?*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
